flatfs.go

```diff
@@ -38,37 +40,53 @@ var _ datastore.Datastore = (*Datastore)(nil)
 
 type ShardFunc func(string) string
 
-func New(path string, getDir ShardFunc, sync bool) (*Datastore, error) {
-	fs := &Datastore{
-		path:   path,
-		getDir: getDir,
-		sync:   sync,
+const IPFS_DEF_SHARD = "/repo/flatfs/shard/v1/next-to-last/2"
+
+func New(path string, fun0 string, sync bool) (*Datastore, error) {
+	fun0 = NormalizeShardFunc(fun0)
+
+	fun1, err := ReadShardFunc(path)
+	if err != nil {
+		return nil, err
 	}
-	return fs, nil
-}
 
-func Prefix(prefixLen int) ShardFunc {
-	padding := strings.Repeat("_", prefixLen)
-	return func(noslash string) string {
-		return (noslash + padding)[:prefixLen]
+	fun := ""
+	switch {
+	case fun0 == "auto" && fun1 == "auto":
+		return nil, fmt.Errorf("shard function not specified")
+	case fun0 == "auto":
+		fun = fun1
+	case fun1 == "auto":
```

Comment:
@whyrusleeping, okay to do things the way you want I will basically have to redo this pull request and the conversion code will require some adjusting.  Let me know.